Extensions/Oxide.CSharp/PluginCompiler.cs

```diff
@@ -494,5 +494,34 @@ private static string GetHash(string filePath, HashAlgorithm algorithm)
                 return BitConverter.ToString(hash).Replace("-", string.Empty).ToLower();
             }
         }
+
+        private static bool FirewallBlocking()
```

Comment:
I don't think an entire method is needed, just check for a valid HTTP code in the existing downloading methods.